scripts/k8s-local-cluster-test/run.sh

```diff
@@ -0,0 +1,160 @@
+#!/bin/bash
+set -eo pipefail
+
+AEMM_URL="amazon-ec2-metadata-mock-service.default.svc.cluster.local"
+AEMM_VERSION="1.2.0"
+AEMM_DL_URL="https://github.com/aws/amazon-ec2-metadata-mock/releases/download/v${AEMM_VERSION}/amazon-ec2-metadata-mock-${AEMM_VERSION}.tgz"
+CLUSTER_NAME_BASE="nth-test"
+DELETE_CLUSTER_ARGS=""
+K8S_VERSION="1.16"
+OVERRIDE_PATH=0
+PRESERVE=false
+PROVISION_CLUSTER_ARGS=""
+START=$(date +%s)
+SCRIPT_PATH="$( cd "$(dirname "$0")" ; pwd -P )"
+TMP_DIR=""
+# VERSION is the source revision that executables and images are built from.
+VERSION=$(git describe --tags --always --dirty || echo "unknown")
+
+function timeout() { perl -e 'alarm shift; exec @ARGV' "$@"; }
+
+function relpath() {
+  perl -e 'use File::Spec; print File::Spec->abs2rel(@ARGV) . "\n"' "${1}" "${2}"
+}
+
+function clean_up {
+    if [[ "$PRESERVE" == false ]]; then
+        "${SCRIPT_PATH}"/delete-cluster.sh $DELETE_CLUSTER_ARGS || :
+        return
+    fi
+    echo "To resume test with the same cluster use: \"-c $TMP_DIR\""""
+}
+
+function exit_and_fail {
+    local pod_id=$(get_nth_worker_pod || :)
+    kubectl logs "${pod_id}" --namespace kube-system || :
+    END=$(date +%s)
+    echo "⏰ Took $(expr "${END}" - "${START}")sec"
+    echo "❌ NTH Integration Test FAILED $CLUSTER_NAME! ❌"
+    exit 1
+}
+
+function get_nth_worker_pod {
+    kubectl get pods -n kube-system \
+      --selector 'app.kubernetes.io/name=aws-node-termination-handler' \
+      --field-selector="spec.nodeName=$CLUSTER_NAME-worker,status.phase=Running" \
+      --sort-by=.metadata.creationTimestamp \
+      --output jsonpath='{.items[-1].metadata.name}'
+}
+
+USAGE=$(cat << 'EOM'
+  Usage: bash run.sh [-p] [-s] [-o] [-b <TEST_BASE_NAME>] [-c <CLUSTER_CONTEXT_DIR>] [-i <AWS Docker image name>] [-s] [-v K8S_VERSION]
+  Executes a test within a provisioned kubernetes cluster with NTH and IMDS pre-loaded.
+
+  Example: bash run.sh -p -s petstore
+
+          Optional:
+            -b          Base name of test (will be used for cluster too)
+            -c          Cluster context directory, if operating on an existing cluster
+            -p          Preserve kind k8s cluster for inspection
+            -i          Provide AWS Service docker image
+            -s          Provide AWS Service name (ecr, sns, sqs, petstore, bookstore)
+            -o          Override path w/ your own kubectl and kind binaries
+            -v          Kubernetes Version (Default: 1.16) [1.14, 1.15, 1.16, 1.17, and 1.18]
+
+EOM
+)
+
+# Process our input arguments
+while getopts "ps:ioc:b:v:" opt; do
```

Comment:
remove `:` after "s" to make the argument required.

Reply:
Removing the colon after "s" resulted in the service tag information not getting picked up and later invalid argument errors